libs/components/forms/src/lib/modules/character-counter/character-counter-screen-reader.pipe.ts

```diff
@@ -0,0 +1,48 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'skyCharacterCounterScreenReader',
+  standalone: true,
+})
+export class SkyCharacterCounterScreenReaderPipe implements PipeTransform {
+  #previousAnnouncementPoint: number | undefined;
+
+  public transform(
+    characterCount: number | undefined,
+    characterCountLimit: number | undefined,
+  ): string {
+    /* Safety check */
+    /* istanbul ignore if */
+    if (characterCount === undefined || characterCountLimit === undefined) {
+      return '';
+    }
+
+    // We want to announce every 10 characters if we are within 50 of the limit or every 50 otherwise.
+    const modulus =
+      characterCountLimit - Math.floor(characterCount / 10) * 10 <= 50
+        ? 10
+        : 50;
+
+    if (
+      characterCount === characterCountLimit ||
+      characterCount % modulus === 0
+    ) {
+      this.#previousAnnouncementPoint = characterCount;
+      return characterCount.toLocaleString();
+    } else if (this.#previousAnnouncementPoint === undefined) {
+      this.#previousAnnouncementPoint = characterCount;
+      return this.#previousAnnouncementPoint.toLocaleString();
+    } else if (
+      Math.floor(characterCount / modulus) ===
```

Comment:
These calculations appear multiple times and happen more than once. Calculate these once outside the `if` statement and assign them to variables to declutter the code and not repeat the same calculations unecessarily.

Reply:
I pulled these out to constants. I did it within the first else and before the if-statement there as a minor efficiency thing. That way - those calculations aren't done unless they are needed for the future logic.